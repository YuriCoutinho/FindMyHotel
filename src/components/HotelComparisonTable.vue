<template>
  <div class="overflow-x-auto">
    <table class="min-w-full bg-white">
      <thead class="bg-emerald-500 text-white">
        <tr>
          <th class="p-3 border border-gray-300 text-left">Hotel</th>
          <th class="p-3 border border-gray-300 text-left">Preço</th>
          <th class="p-3 border border-gray-300 text-left">Estrelas</th>
          <th class="p-3 border border-gray-300 text-left">Comodidades</th>
          <th class="p-3 border border-gray-300 text-left">Ação</th>
        </tr>
      </thead>
      <tbody>
        <tr v-for="hotel in hotels" :key="hotel.id" class="hover:bg-gray-100">
          <td class="p-3 border border-gray-300">{{ hotel.name }}</td>
          <td class="p-3 border border-gray-300">R$ {{ hotel.price }}</td>
          <td class="p-3 border border-gray-300">{{ hotel.stars }}</td>
          <td class="p-3 border border-gray-300">{{ hotel.amenities.join(', ') }}</td>
          <td class="p-3 border border-gray-300">
            <button @click="$emit('remove', hotel.id)" class="text-red-500 hover:text-red-700">
              Remover
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</template>

<script setup lang="ts">
interface Hotel {
  id: string
  name: string
  price: number
  stars: number
  amenities: string[]
}

defineProps<{
  hotels: Hotel[]
}>()
</script>
