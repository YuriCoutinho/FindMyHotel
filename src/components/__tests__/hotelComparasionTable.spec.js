import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import HotelComparisonTable from '@/components/HotelComparisonTable.vue'

describe('HotelTable.vue', () => {
  const mockHotels = [
    {
      id: '1',
      name: 'Hotel Example 1',
      price: 150,
      stars: 4,
      amenities: ['Wi-Fi', 'Breakfast', 'Pool']
    },
    {
      id: '2',
      name: 'Hotel Example 2',
      price: 250,
      stars: 5,
      amenities: ['Wi-Fi', 'Spa', 'Gym']
    }
  ]

  const createWrapper = (hotels = mockHotels) => {
    return mount(HotelComparisonTable, {
      props: { hotels }
    })
  }

  it('Should render the component', () => {
    const wrapper = createWrapper()
    expect(wrapper.exists()).toBe(true)
  })

  it('Should display hotel information correctly', () => {
    const wrapper = createWrapper()
    const rows = wrapper.findAll('tbody tr')
    expect(rows.length).toBe(mockHotels.length)

    mockHotels.forEach((hotel, index) => {
      const columns = rows[index].findAll('td')
      expect(columns[0].text()).toBe(hotel.name)
      expect(columns[1].text()).toContain(hotel.price.toString())
      expect(columns[2].text()).toBe(hotel.stars.toString())
      expect(columns[3].text()).toBe(hotel.amenities.join(', '))
    })
  })

  it('Should emit "remove" event with hotel id when remove button is clicked', async () => {
    const wrapper = createWrapper()
    const removeButton = wrapper.findAll('button').at(0)

    await removeButton.trigger('click')

    expect(wrapper.emitted('remove')).toHaveLength(1)
    expect(wrapper.emitted('remove')[0]).toEqual([mockHotels[0].id])
  })

  it('Should match the snapshot', () => {
    const wrapper = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
