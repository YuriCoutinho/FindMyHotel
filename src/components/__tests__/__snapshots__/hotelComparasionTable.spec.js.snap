// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HotelTable.vue > Should match the snapshot 1`] = `
"<div class="overflow-x-auto">
  <table class="min-w-full bg-white">
    <thead class="bg-emerald-500 text-white">
      <tr>
        <th class="p-3 border border-gray-300 text-left">Hotel</th>
        <th class="p-3 border border-gray-300 text-left">Preço</th>
        <th class="p-3 border border-gray-300 text-left">Estrelas</th>
        <th class="p-3 border border-gray-300 text-left">Comodidades</th>
        <th class="p-3 border border-gray-300 text-left">Ação</th>
      </tr>
    </thead>
    <tbody>
      <tr class="hover:bg-gray-100">
        <td class="p-3 border border-gray-300">Hotel Example 1</td>
        <td class="p-3 border border-gray-300">R$ 150</td>
        <td class="p-3 border border-gray-300">4</td>
        <td class="p-3 border border-gray-300">Wi-Fi, Breakfast, Pool</td>
        <td class="p-3 border border-gray-300"><button class="text-red-500 hover:text-red-700"> Remover </button></td>
      </tr>
      <tr class="hover:bg-gray-100">
        <td class="p-3 border border-gray-300">Hotel Example 2</td>
        <td class="p-3 border border-gray-300">R$ 250</td>
        <td class="p-3 border border-gray-300">5</td>
        <td class="p-3 border border-gray-300">Wi-Fi, Spa, Gym</td>
        <td class="p-3 border border-gray-300"><button class="text-red-500 hover:text-red-700"> Remover </button></td>
      </tr>
    </tbody>
  </table>
</div>"
`;
